pkg/controller/deployment/deployment_controller.go

```diff
@@ -424,6 +424,27 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
+// syncPausedDeployment allows scaling of the deployment (either done manually
+// or by autoscalers) to propagate to the active rc of the deployment.
+func (dc *DeploymentController) syncPausedDeployment(deployment extensions.Deployment) error {
+	newRC, err := deploymentutil.GetNewRCFromList(deployment, dc.client,
+		func(namespace string, options api.ListOptions) ([]api.ReplicationController, error) {
+			return dc.rcStore.List()
+		})
+	if err != nil {
+		return err
+	}
+	if newRC == nil {
+		// There is no replication controller that matches the intent of
+		// this deployment which means that either the deployment has never
+		// run (created as paused), or it was switched to paused and has been
+		// modified since. Such is life.
+		return nil
+	}
+	_, err = dc.scaleRCAndRecordEvent(newRC, deployment.Spec.Replicas, deployment)
```

Comment:
I don't think the un-paused scaling behavior is what we want, either. If the new scale is greater than the scale of the newest RC, then it scales down just the newest RC. However, that doesn't take into account that older RCs may still have replicas. Scaling proportionally makes sense to me while un-paused, also.

It also doesn't take into account which pods are ready or not ready, but that's another issue.